autobahn/src/main/java/io/crossbar/autobahn/wamp/types/IEndpoint.java

```diff
@@ -0,0 +1,10 @@
+package io.crossbar.autobahn.wamp.types;
+
+import java.util.List;
+import java.util.Map;
+
+@FunctionalInterface
```

Comment:
This basically allows us to pass in anonymous methods as parameters; new in Java8. Any method having this as a parameter will ensure that the callback method that we pass in have exactly the same signature.

In this case it helps our API to only except a method as callback that has the signature
`List<Object> args, Map<String, Object> kwargs`